hmacauth.go

```diff
@@ -5,35 +5,100 @@ import (
 	"crypto"
 	"crypto/hmac"
 	"encoding/base64"
+	"errors"
 	"net/http"
 	"strconv"
 	"strings"
 )
 
-var supportedAlgorithms map[string]crypto.Hash
+var supportedAlgorithms map[string]crypto.Hash = map[string]crypto.Hash{
+	"md4":       crypto.MD4,
+	"md5":       crypto.MD5,
+	"sha1":      crypto.SHA1,
+	"sha224":    crypto.SHA224,
+	"sha256":    crypto.SHA256,
+	"sha384":    crypto.SHA384,
+	"sha512":    crypto.SHA512,
+	"ripemd160": crypto.RIPEMD160,
+}
+
 var algorithmName map[crypto.Hash]string
 
 func init() {
-	supportedAlgorithms = map[string]crypto.Hash{
-		"sha1": crypto.SHA1,
-	}
-
 	algorithmName = make(map[crypto.Hash]string)
 	for name, algorithm := range supportedAlgorithms {
 		algorithmName[algorithm] = name
+		if algorithm.Available() == false {
```

Comment:
My understanding is that IF your code is linking to a version of a crypto library when building this service/library (as HASH implementations grow over time), you want to verify against the requestors use of a HASH function so that you can perform the same function to reproduce the HASH such as SHA1, SHA2, SHA3, whereby you may not have SHA3 implementation support yet.